_posts/2021-06-28-sparklyr-1.7.0-released/sparklyr-1.7.0.Rmd

```diff
@@ -0,0 +1,386 @@
+---
+title: "sparklyr 1.7: new data sources and spark_apply() capabilities, better interfaces for sparklyr extensions, etc"
+description: |
+  Sparklyr 1.7 delivers much anticipated improvements such as R interfaces for image and binary data sources, several new spark_apply() capabilities, better integration with sparklyr extensions, among others.
+author:
+  - name: Yitao Li
+    url: https://github.com/yitao-li
+    affiliation: RStudio
+    affiliation_url: https://www.rstudio.com
+slug: sparklyr-1.7
+date: 06-28-2021
+bibliography: bibliography.bib
+categories:
+  - R
+  - Packages/Releases
+  - Distributed Computing
+output:
+  distill::distill_article:
+    self_contained: false
+preview: images/sparklyr-1.7.png
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = FALSE)
+```
+
+[`Sparklyr`](https://sparklyr.ai) 1.7 is now available on [CRAN](https://cran.r-project.org/web/packages/sparklyr/index.html)!
+
+To install `sparklyr` 1.7 from CRAN, run
+
+```{r echo=TRUE, eval=FALSE}
+install.packages("sparklyr")
+```
+
+In this blog post, we wish to present the following highlights from the `sparklyr` 1.7 release:
+
+* [Image and binary data sources](#image-and-binary-data-sources)
+* [New spark_apply() capabilities](#new-spark_apply-capabilities)
+* [Better integration with sparklyr extensions](#better-integration-with-sparklyr-extensions)
+* [Other exciting news](#other-exciting-news)
+
+## Image and binary data sources
+
+As a unified analytics engine for large-scale data processing, [Apache Spark](https://spark.apache.org)
+is well-known for its ability to tackle challenges associated with the volume, velocity, and last but
+not least, the variety of big data. Therefore it is hardly surprising to see in response to recent
+advances in deep learning frameworks, Apache Spark has introduced the built-in support for
+[image data source](https://issues.apache.org/jira/browse/SPARK-22666) in its version 2.4 release, and
+same for [binary data source](https://issues.apache.org/jira/browse/SPARK-25348) in its 3.0 release.
+The corresponding R interfaces for both data sources in `sparklyr`, namely,
+[`spark_read_image()`](https://spark.rstudio.com/reference/spark_read_image.html) and
+[`spark_read_binary()`](https://spark.rstudio.com/reference/spark_read_binary.html), were shipped
+recently as part of `sparklyr` 1.7.
+
+The usefulness of data source functionalities such as `spark_read_image()` is perhaps best illustrated
+by a quick demo below, where image files are read into Spark dataframes and transformed into highly
```

Comment:
Building something that potentially defeats CAPTCHA so rapidly with a handful of open-source bits and pieces and without having to implement any of the crucial components from scratch has been a lot of fun :slightly_smiling_face: 

Maybe we should make this demo its own separate blog post and link that post here? Everything should still be within the same PR though, as the blog post about this demo depends on `sparklyr` 1.7, but we have not announced the release of `sparklyr` 1.7 yet publicly.